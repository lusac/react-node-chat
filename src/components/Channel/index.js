import React from 'react'
import PropTypes from 'prop-types'

export default class Channel extends React.Component {
  constructor(props) {
    super()
    this.state = {
      msg: '',
      channel: props.channel || {},
      msgs: props.channel.msgs || []
    }
  }

  componentDidUpdate(prevProps) {
    if (this.props.channel.id !== prevProps.channel.id) {
      this.updateChannelConnection()
    }
  }

  updateChannelConnection() {
    this.setState({
      msgs: this.props.channel.msgs
    }, () => {
      this.props.socket.off('message').on('message', (msg) => {
        this.setState({
          msgs: [...this.state.msgs, msg]
        })
      })
    })
  }

  onMsgChange(e) {
    this.setState({
      msg: e.target.value
    })
  }

  onMsgSubmit(e) {
    e.preventDefault()
    this.props.socket.emit('message', {
      channelID: this.props.channel.id,
      msg: this.state.msg
    })
    this.setState({
      msg: ''
    })
  }

  render() {
    return (
      <div className="channel">
        <div className="channel__header">
          header
        </div>

        <div className="channel__section">
          <div className="channel__chat">
            {!this.props.channel.id && <span>Nenhum canal selecionado</span>}
            {this.props.channel.id && this.state.msgs.map((msg, i) => <div key={i}>{msg}</div> )}
          </div>

          {this.props.channel.id &&
            <div className="channel__chat-input__wrapper">
              <form onSubmit={this.onMsgSubmit.bind(this)}>
                <input
                  placeholder="escreva uma mensagem"
                  className="channel__chat-input"
                  value={this.state.msg}
                  onChange={this.onMsgChange.bind(this)} />
                {/* <input type="submit" value="Submit" /> */}
              </form>
            </div>
          }
        </div>
      </div>
    )
  }
}

Channel.propTypes = {
  channel: PropTypes.object,
  socket: PropTypes.object
}