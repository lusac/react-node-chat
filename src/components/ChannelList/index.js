import React from 'react'
import PropTypes from 'prop-types'
import NewChannelModal from '../NewChannelModal'

export default class ChannelList extends React.Component {
  constructor() {
    super()
    this.state = {
      showModal: false
    }
  }

  selectChat(id) {
    this.props.socket.emit('join channel', id)
  }

  toggleModal() {
    this.setState({
      showModal: !this.state.showModal
    })
  }

  render() {
    return (
      <div className="channel-list">
        <div className="channel-list__header">
          Ol√° <strong className="channel-list__header__name">{this.props.username}</strong>!
        </div>

        <div className="channel-list__section">
          <div className="channel-list__section__title">
            <span>Canais</span>
            <span
              className="plus"
              onClick={this.toggleModal.bind(this)}>+</span>
          </div>

          {Object.keys(this.props.channels).map(id => {
            return (
              <span
                key={id}
                className={"channel-list__item " + (this.props.channel.id === id ? 'selected' : '')}
                onClick={this.selectChat.bind(this, id)}>
                  # {this.props.channels[id].name}</span>
            )
          })}
        </div>

        {this.state.showModal &&
          <NewChannelModal
            toggleModal={this.toggleModal.bind(this)}
            socket={this.props.socket} />
        }
      </div>
    )
  }
}

ChannelList.propTypes = {
  socket: PropTypes.object,
  channel: PropTypes.object,
  username: PropTypes.string,
  channels: PropTypes.object,
};