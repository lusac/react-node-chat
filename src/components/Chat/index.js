import React from 'react';

class Chat extends React.Component {
  render() {
    return (
      <div className="chat-room">
        <div className="chat-room__messages">
          oie
        </div>

        <div className="chat-room__input__wrapper">
          <input placeholder="escreva uma mensagem" className="chat-room__input"/>
        </div>
      </div>
    )
  }
}

export default Chat;
