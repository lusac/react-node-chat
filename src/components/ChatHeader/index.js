import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

export class ChatHeader extends React.Component {
  render() {
    return (
      <div className="chat-header">
        <div className="chat-header__profile">
            Ol√° {this.props.username}!
        </div>

        {this.props.chat.id &&
          <div className="chat-header__chat-name">chat {this.props.chat.id}</div>
        }
      </div>
    )
  }
}

ChatHeader.propTypes = {
  username: PropTypes.string,
  chat: PropTypes.object
};

function mapStateToProps(state) {
  return ({
    chat: state.chat
  })
}

export default connect(mapStateToProps, {})(ChatHeader)


