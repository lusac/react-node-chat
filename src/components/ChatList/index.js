import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { getChats } from '../../actions/chats'
import { getChat } from '../../actions/chat'

export class ChatList extends React.Component {
  componentDidMount() {
    this.props.getChats()
  }

  selectChat(id) {
    if (this.props.socket.socket) {
      this.props.getChat(id)
      this.props.socket.socket.emit('room', id)
    }
  }

  render() {
    return (
      <div className="chat-list">
        {this.props.chats.loading && <span>loading...</span>}
        {!this.props.chats.loading && this.props.chats.chats.map((room) => {
          return <span key={room.id} className="chat-list__item" onClick={this.selectChat.bind(this, room.id)}>{room.name}</span>
        })}
      </div>
    )
  }
}

ChatList.propTypes = {
  chats: PropTypes.object,
  getChats: PropTypes.func,
  getChat: PropTypes.func,
  socket: PropTypes.object
};

function mapStateToProps(state) {
  return ({
    chat: state.chat,
    chats: state.chats,
    socket: state.socket
  })
}

export default connect(mapStateToProps, {
  getChats,
  getChat
})(ChatList)
