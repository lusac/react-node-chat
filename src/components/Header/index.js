import React from 'react'

export const Header = () => {
  return (
    <div className="header">
      <label
        htmlFor="toggleChannelList"
        className="burguer">
        <span></span>
        <span></span>
        <span></span>
      </label>
    </div>
  )
}