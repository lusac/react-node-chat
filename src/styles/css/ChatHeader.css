.chat-header {
  height: 50px;
  color: #FFF;
  padding: 10px;
  display: grid;
  grid-area: header;
  align-items: center;
  background: cadetblue;
  grid-template-areas: "header-profile header-chat-name header-chat-name";
}
.chat-header__profile {
  grid-area: header-profile;
}
.chat-header__chat-name {
  grid-area: header-chat-name;
}

/*# sourceMappingURL=ChatHeader.css.map */
