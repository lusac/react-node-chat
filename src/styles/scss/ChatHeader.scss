.chat-header {
  height: 50px;
  color: #FFF;
  padding: 10px;
  display: grid;
  grid-area: header;
  align-items: center;
  background: cadetblue;
  grid-template-areas: "header-profile header-chat-name header-chat-name";

  &__profile {
    grid-area: header-profile;
  }

  &__chat-name {
    grid-area: header-chat-name;
  }
}