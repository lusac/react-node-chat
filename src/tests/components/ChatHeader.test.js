/* global it describe expect beforeEach */
import React from 'react';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';
import { ChatHeader } from '../../components/ChatHeader'

configure({ adapter: new Adapter() })

describe('ChatHeader component', () => {
  describe('snapshot', () => {
    it('without chat title', () => {
      const chat = { loading: true }
      const renderedValue =  renderer.create(<ChatHeader username='xunda' chat={chat} />).toJSON()
      expect(renderedValue).toMatchSnapshot();
    })

    it('with chat title', () => {
      const chat = { loading: false, id: 1 }
      const renderedValue =  renderer.create(<ChatHeader username='xunda' chat={chat} />).toJSON()
      expect(renderedValue).toMatchSnapshot();
    })
  })

  describe('rendered', () => {
    describe('without chat title', () => {
      let wrapper, chat = {}

      beforeEach(() => {
        wrapper = shallow(<ChatHeader username='xunda' chat={chat} />)
      })

      it('correct username', () => {
        expect(wrapper.find('.chat-header__profile').get(0).props.children.join('')).toBe('Olá xunda!')
      })

      it('no chat title', () => {
        expect(wrapper.find('.chat-header__chat-name').get(0)).toBeUndefined()
      })
    })

    describe('with chat title', () => {
      let wrapper, chat = { id: 1 }

      beforeEach(() => {
        wrapper = shallow(<ChatHeader username='xunda' chat={chat} />)
      })

      it('correct username', () => {
        expect(wrapper.find('.chat-header__profile').get(0).props.children.join('')).toBe('Olá xunda!')
      })

      it('correct chat title', () => {
        expect(wrapper.find('.chat-header__chat-name').get(0).props.children.join('')).toBe('chat 1')
      })
    })
  })
})