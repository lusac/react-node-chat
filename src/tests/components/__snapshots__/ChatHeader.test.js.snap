// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHeader component snapshot with chat title 1`] = `
<div
  className="chat-header"
>
  <div
    className="chat-header__profile"
  >
    Olá 
    xunda
    !
  </div>
  <div
    className="chat-header__chat-name"
  >
    chat 
    1
  </div>
</div>
`;

exports[`ChatHeader component snapshot without chat title 1`] = `
<div
  className="chat-header"
>
  <div
    className="chat-header__profile"
  >
    Olá 
    xunda
    !
  </div>
</div>
`;
